OpenRA.Mods.Common/Traits/GainsExperience.cs

```diff
@@ -19,7 +19,7 @@ namespace OpenRA.Mods.Common.Traits
 	[Desc("This actor's experience increases when it has killed a GivesExperience actor.")]
 	public class GainsExperienceInfo : ITraitInfo, Requires<ValuedInfo>, Requires<UpgradeManagerInfo>
 	{
-		[FieldLoader.Require]
+		[FieldLoader.Require, UpgradeGrantedReference, UpgradeUsedReference] // HACK: Treat upgrades as used
```

Comment:
@penev92 we define upgrades in the actor defaults which are not necessarily implemented by the traits on every actor.